ext/xsl/tests/xsltprocessor_registerPHPFunctions-array-notallowed.phpt

```diff
@@ -19,7 +19,7 @@ var_dump($proc->transformToXml($dom));
 NULL
 
 Warning: XSLTProcessor::transformToXml(): Not allowed to call handler 'ucwords()' in %s on line %d
-NULL
```

Comment:
So the null return value was related to the warning, wasn't it? As far as I see, `xsltRegisterExtModuleFunction()` triggers the warning via the `xsl_ext_function_string_php` callback. Wouldn't it make more sense to return `false` in this case then?

Reply:
I think the NULL return is more of a side effect here because there are no children in the result. I think that if there were a warning, and children were returned that we would get a string with contents here. So in general I don't think we can return false?

Reply:
> I think the NULL return is more of a side effect here because there are no children in the result. I think that if there were a warning, and children were returned that we would get a string with contents here.

Aha, I see :) Then the problem is with these error cases casting the result to an empty string via `valuePush(ctxt, xmlXPathNewString((const xmlChar *) ""));`). So ideally these should have been errors in my opinion.

Reply:
They are warnings now, so we don't expect false to be returned. If they were errors/exceptions we could indeed get rid of those pushes, but if we get rid of them now we get a stack error for xsl.
I'm actually starting to doubt any of these changes are a good idea. Although returning NULL could be a footgun, it does mimick the behaviour of the C counterpart of the function.

Reply:
Yeah, this problem is very convoluted, and I don't feel like I'm able to make an informed decision :(